### PR TITLE
Adds in example of using collections

### PR DESCRIPTION
**Pull request by warrenbuckley**
Adds in a code example for using collections.
Worth sanity checking @zpqrtbnk please

Also technical docs reviewer @marcemarc can you ensure my code sample reads fine, grammar & the example makes sense, gets the point across on what can be done & feel free to tweak/change if you feel it can be improved :)

Happy Easter all :)

### CODE REVIEW

**Comment by jmayntzhusen** [comment]
This is quite nice @warrenbuckley I wanted to do a spot check on some name spaces since I haven't worked with this at all and am not familiar Composing, and I learned there is both a CollectionBuilderBase class and a BuilderCollectionBase class 😅

Will leave the review for these other guys where this makes sense 😁 happy Easter!

File: Implementation/Composing/index.md [hunk 1]
```diff
@@ -4,18 +4,18 @@ versionFrom: 8.0.0
 ---
 
 # Composing
-Customising the behaviour of an Umbraco Application at 'start up'. eg adding, removing or replacing the core functionality of Umbraco or registering custom code to subscribe to events. 
+Customising the behaviour of an Umbraco Application at 'start up'. eg adding, removing or replacing the core functionality of Umbraco or registering custom code to subscribe to events.
 
 ## Overview
-An Umbraco application is a `Composition` made of many different 'collections' of specific functionality/implementation logic/components (eg. UrlProviders, ContentFinders - see below for a full list). These collections are populated when the Umbraco Application starts up. 
+An Umbraco application is a `Composition` made of many different 'collections' of specific functionality/implementation logic/components (eg. UrlProviders, ContentFinders - see below for a full list). These collections are populated when the Umbraco Application starts up.
```
Comment (zpqrtbnk) [inline Implementation/Composing/index.md]: composition is not only about "collections", you can compose single items such as the "last chance content finder" - there is only one

File: Implementation/Composing/index.md [hunk 2]
```diff
@@ -4,18 +4,18 @@ versionFrom: 8.0.0
...
 
 'Composing' is the term used to describe the process of curating which pieces of functionality should be included in a particular collection. The code that implements these choices at start up is called a `Composer`.
 
-A `Component`, is a generic wrapper for writing custom code during composition, it has two methods: `Initialize()` and `Terminate()` and these are executed when the Umbraco Application starts up, and when it shuts down, respectively. Typically a `Component` may be used to wire up custom code to handle a particular event in Umbraco. (see content saving example below). 
+A `Component`, is a generic wrapper for writing custom code during composition, it has two methods: `Initialize()` and `Terminate()` and these are executed when the Umbraco Application starts up, and when it shuts down, respectively. Typically a `Component` may be used to wire up custom code to handle a particular event in Umbraco. (see content saving example below).
```
Comment (zpqrtbnk) [inline Implementation/Composing/index.md]: a component is not to run "during composition" - it initalizes during composition, but can be used to run code (event handlers...) while the app runs - ie it keeps "running" as long as umbraco is not terminated

File: Implementation/Composing/index.md [hunk 3]
```diff
@@ -157,7 +157,7 @@ Is an implementation of IUserComposer, that provides a quicker way to add a cust
 ```csharp
         [RuntimeLevel(MinLevel = RuntimeLevel.Run)]
```
Comment (zpqrtbnk) [inline Implementation/Composing/index.md]: rather that use a `RuntimeLevel` attribute, just implement `IUserComposer` that's cleaner

File: Implementation/Composing/index.md [hunk 4]
```diff
@@ -352,3 +352,181 @@ The runtime has detected an Umbraco install which needed to be upgraded, and is
...
+}
+```
+
+## Example of using Lazy Collections with Type Scanning
+
+You may wish to create an Umbraco package that allows package consumers to extend and add additional functionality. In this example we show how you can use the `LazyCollectionBuilderBase` to scan assemblies that implement your interface by using the `TypeLoader`
+
+:::warning
+Add types from assemblies - be conscious of doing type scanning, as this adds time to bootup of Umbraco.
+If you still need to use type scanning, ensure your Interface implements `IDiscoverable`
```
Comment (zpqrtbnk) [inline Implementation/Composing/index.md]: should explain _why_ ppl need to implement `IDiscoverable`

File: Implementation/Composing/index.md (hunk 2 above)
Comment (warrenbuckley) [reply Implementation/Composing/index.md]: I think this is already mentioned as by this part @zpqrtbnk 

>Typically a `Component` may be used to wire up custom code to handle a particular event in Umbraco. (see content saving example below).

**Comment by warrenbuckley** [comment]
@jmayntzhusen think this should be ready for review from you. As think I done all the suggestions from @zpqrtbnk

**Comment by jmayntzhusen** [comment]
That's great, will merge. Thanks for updating the docs! 😄

**Pull request merged**